### PR TITLE
feat: new publish wizard publication fee step

### LINKED ISSUE #2203: [Publish Wizard] Move the Publication fee to the last step
## 🎉 Description

According to the new Wizard, the publication fee detail is the last step.
Tasks:
- Move the current version to be the latest step
- Adapt the current version to the new designs
- The button should read "Pay XXX MANA"

* [ ] 🐛 This is a bug report.
* [X] 📈 This is a feature request.

### LINKED ISSUE #2204: [Publish Wizard] Congratulations step
## 🎉 Description

There's a new step at the end, the congratulations one.
Tasks:
- Add this new step and the content that the designs show
- The button should send the user to the forum post

* [ ] 🐛 This is a bug report.
* [X] 📈 This is a feature request.

### PR DESCRIPTION
**Pull request by cyaiox**
This PR adds the pay for the publication fee and the congratulations step.

Screen Shot 2022-10-28 at 09 58 02

Screen Shot 2022-10-28 at 10 22 48

Closes #2203
Closes #2204

### CODE REVIEW

File: src/modules/collection/reducer.ts [hunk 1]
```diff
@@ -227,7 +227,8 @@ export function collectionReducer(state: CollectionState = INITIAL_STATE, action
             ...state.data[collection.id],
             lock: collection.lock
           }
-        }
+        },
+        loading: loadingReducer(state.loading, action)
```
Comment (cyaiox) [inline src/modules/collection/reducer.ts:231]: this line is to update the loading state when publishing a collection is successful

File: src/modules/collection/reducer.ts [hunk 2]
```diff
@@ -261,7 +262,8 @@ export function collectionReducer(state: CollectionState = INITIAL_STATE, action
             ...state.data[collection.id],
             forumLink
           }
-        }
+        },
+        loading: loadingReducer(state.loading, action)
```
Comment (cyaiox) [inline src/modules/collection/reducer.ts:266]: this line is to update the loading state when creating the forum post for a collection is successful

File: src/components/Modals/PublishWizardCollectionModal/ConfirmCollectionItemsStep/ConfirmCollectionItemsStep.tsx [hunk 3]
```diff
@@ -70,17 +70,15 @@ export const ConfirmCollectionItemsStep: React.FC<{ items: Item[]; onNextStep: (
         <Column>
           <Row className="details">
             <Column grow={true}>
-              <p className="title">{t('publish_collection_modal_with_oracle.confirm_collection_items_step.title')}</p>
-              <p className="subtitle">
-                {t('publish_collection_modal_with_oracle.confirm_collection_items_step.subtitle', { enter: <br /> })}
-              </p>
-              <p className="description">{t('publish_collection_modal_with_oracle.confirm_collection_items_step.description')}</p>
+              <p className="title">{t('publish_wizard_collection_modal.confirm_collection_items_step.title')}</p>
```
Comment (cyaiox) [inline src/components/Modals/PublishWizardCollectionModal/ConfirmCollectionItemsStep/ConfirmCollectionItemsStep.tsx:73]: updated the translation path

File: src/components/Modals/PublishWizardCollectionModal/ConfirmCollectionNameStep/ConfirmCollectionNameStep.tsx [hunk 4]
```diff
@@ -25,23 +25,21 @@ export const ConfirmCollectionNameStep: React.FC<{ collection: Collection; onNex
         <Column>
           <Row className="details">
             <Column grow={true}>
-              <p className="title">{t('publish_collection_modal_with_oracle.confirm_collection_name_step.title')}</p>
-              <p className="subtitle">{t('publish_collection_modal_with_oracle.confirm_collection_name_step.subtitle')}</p>
-              <p className="description">{t('publish_collection_modal_with_oracle.confirm_collection_name_step.description')}</p>
+              <p className="title">{t('publish_wizard_collection_modal.confirm_collection_name_step.title')}</p>
```
Comment (cyaiox) [inline src/components/Modals/PublishWizardCollectionModal/ConfirmCollectionNameStep/ConfirmCollectionNameStep.tsx:28]: updated the translation path

File: src/components/Modals/PublishWizardCollectionModal/CongratulationsStep/CongratulationsStep.tsx [hunk 5]
```diff
@@ -0,0 +1,41 @@
...
+import { Button, Column, Modal, Row } from 'decentraland-ui'
+import { t } from 'decentraland-dapps/dist/modules/translation/utils'
+import { Collection } from 'modules/collection/types'
+import './CongratulationsStep.css'
+
+export const CongratulationsStep: React.FC<{ collection: Collection }> = props => {
+  const { collection } = props
+
+  return (
+    <>
```
Comment (juanmahidalgo) [inline src/components/Modals/PublishWizardCollectionModal/CongratulationsStep/CongratulationsStep.tsx:11]: nitpick: I believe we don't need the fragment here, right? 
```suggestion
    <Modal.Content className="CongratulationsStep">
```

File: src/components/Modals/PublishWizardCollectionModal/PayPublicationFeeStep/PayPublicationFeeStep.tsx [hunk 6]
```diff
@@ -0,0 +1,135 @@
...
+
+  let priceUSD = '0'
+  let totalPrice = '0'
+  let totalPriceUSD = '0'
+  let hasInsufficientMANA = true
+
+  if (refRarity) {
+    priceUSD = (refRarity as any).prices!.USD
+
+    totalPrice = ethers.BigNumber.from((refRarity as any).prices!.MANA)
```
Comment (juanmahidalgo) [inline src/components/Modals/PublishWizardCollectionModal/PayPublicationFeeStep/PayPublicationFeeStep.tsx:27]:
```suggestion
    totalPrice = ethers.BigNumber.from(refRarity.prices!.MANA)
```
same from above

File: src/components/Modals/PublishWizardCollectionModal/PayPublicationFeeStep/PayPublicationFeeStep.tsx [hunk 7]
```diff
@@ -0,0 +1,135 @@
...
+      .mul(items.length)
+      .toString()
+
+    hasInsufficientMANA = !!wallet && wallet.networks.MATIC.mana < Number(ethers.utils.formatEther(totalPrice))
+  }
+
+  const hasCollectionError = unsyncedCollectionError || collectionError
+
+  return (
+    <>
```
Comment (juanmahidalgo) [inline src/components/Modals/PublishWizardCollectionModal/PayPublicationFeeStep/PayPublicationFeeStep.tsx:41]: same, I think we don't need the fragment here

**Review by juanmahidalgo — APPROVED** [review]
Left a few small comments but looks great 👍

**Pull request merged**